dip-0022.md

```diff
@@ -0,0 +1,94 @@
+<pre>
+  DIP: 0022
+  Title: Enhanced Hard Fork Mechanism
+  Author(s): Pasta
+  Special-Thanks:
+  Comments-Summary: No comments yet.
+  Status: Proposed
+  Type: Standard
+  Created: 2021-07-29
+  License: MIT License
+</pre>
+
+## Abstract
+
+This DIP aims to introduce an improved hard fork activation mechanism that includes both masternodes
+and miners while also ensuring sufficient upgrade time is available to partners.
+
+## Motivation
+
+Currently, we utilize a heavily modified version of BIP-9 for hard fork activation in Dash Core. In
+the previous version of Dash Core, this calculation took into account the percentage of both miners
+and masternodes that had upgraded. However, since the introduction of the Deterministic Masternode
+List, the activations have purely been based on miner signalling.
+
+## Previous Work
+
+* [BIP-0009: Version bits with timeout and
+  delay](https://github.com/bitcoin/bips/blob/master/bip-0009.mediawiki)
+* [Dash dynamic activation thresholds](https://github.com/dashpay/dash/pull/3692)
+
+## Current System
```

Comment:
Will this result in adding new status values to [versionbits](https://github.com/dashpay/dash/blob/master/src/versionbits.h#L20-L26)? If so it might be worth mentioning.